ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
```

Comment:
It's an overridable default. I kind of like having it in here to describe the default operating behaviour. I can't see it causing problems. Also avoids third party dependency frameworks that do the same but with more boilerplate/learning.